packages/driver/cypress/integration/commands/net_stubbing_spec.ts

```diff
@@ -594,6 +594,61 @@ describe('network stubbing', { retries: 2 }, function () {
         .wait('@foo').its('response.body').should('eq', 'replaced the body')
       })
     })
+
+    // @see https://github.com/cypress-io/cypress/issues/9599
+    context('cors preflight', function () {
+      const corsUrl = 'http://diff.foobar.com:3501/no-cors'
+
+      before(() => {
+        cy.visit('http://127.0.0.1:3500/fixtures/dom.html')
+      })
+
+      it('responds to OPTIONS requests', function () {
+        cy.request({ method: 'OPTIONS', url: corsUrl })
```

Comment:
wait, doesn't it fail on this request without an option to allow?

Reply:
this is doing a sanity check that an un-stubbed request does not have a successful OPTIONS response

Reply:
Yes, but wouldn't a `cy.request` to a non-existent URL just fail without `failOnStatusCode: false`? I don't know how this test is passing

Reply:
`/no-cors` ends the request successfully but without setting any type of CORS headers: https://github.com/cypress-io/cypress/blob/61cd21eb4718d4167e8d9f521c44d36ab2673657/packages/driver/cypress/plugins/server.js#L21-L23

so for the first request, it will be missing a valid CORS preflight response and will fail here: https://github.com/cypress-io/cypress/blob/61cd21eb4718d4167e8d9f521c44d36ab2673657/packages/driver/cypress/integration/commands/net_stubbing_spec.ts#L612-L618

but then, we `cy.intercept` the same URL and try again, and the same ajax request succeeds:

https://github.com/cypress-io/cypress/blob/61cd21eb4718d4167e8d9f521c44d36ab2673657/packages/driver/cypress/integration/commands/net_stubbing_spec.ts#L620-L628

...these would make great comments in the test, i'll add 'em